entrypoint.sh

```diff
@@ -0,0 +1,147 @@
+  -E xpack.security.http.ssl.certificate_authorities=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/ca/ca.crt \
+  -E xpack.security.transport.ssl.enabled=true \
+  -E xpack.security.transport.ssl.verification_mode=certificate \
+  -E xpack.security.transport.ssl.certificate=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/instance/instance.crt \
+  -E xpack.security.transport.ssl.key=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/instance/instance.key \
+  -E xpack.security.transport.ssl.certificate_authorities=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/instance/instance.crt \
+  -E xpack.license.self_generated.type=basic & 
+
+while [ "$password_reset_elastic" = false ]; do
+  if check_elasticsearch_running; then
+    sleep 5;
+    ELASTIC_RANDOM_PASSWORD=$(
+      elasticsearch-${EK_VERSION}/bin/elasticsearch-reset-password \
+      -v \
+      --url "https://localhost:9200" \
+      -u elastic \
+      -b \
+      -E xpack.security.http.ssl.enabled=true \
+      -E xpack.security.http.ssl.certificate=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/instance/instance.crt \
+      -E xpack.security.http.ssl.key=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/instance/instance.key \
+      -E xpack.security.http.ssl.certificate_authorities=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/ca/ca.crt
+    );
+    exit_code=$?
+    if [ $exit_code -eq 0 ]; then
+
+      # Generate new password and grab the value from 'awk'
+      ELASTIC_NEW_PASSWORD=$(echo "$ELASTIC_RANDOM_PASSWORD" | awk '/New value:/ {print $3}');    
+
+      clear
+      echo "----------------------------------------------------------------------------------"
+      echo "------------------- RESETTING 'elastic' USER PASSWORD ----------------------------"
+      echo "----------------------------------------------------------------------------------"
+      echo "Password reset successful. '$ELASTIC_NEW_PASSWORD'"
+      echo "----------------------------------------------------------------------------------"
+
+      # Flip global check value to stop this loop
+      password_reset_elastic=true  
+    fi
+  else
+    sleep 1
+  fi
+done &
+
+# Start Kibana in SSL mode
+kibana-${EK_VERSION}/bin/kibana \
+  -Q \
+  --allow-root \
+  --host 0.0.0.0 \
+  --server.ssl.enabled=true \
+  --server.ssl.certificate=/home/elastic/kibana-${EK_VERSION}/instance/instance.crt \
+  --server.ssl.key=/home/elastic/kibana-${EK_VERSION}/instance/instance.key \
+  --elasticsearch.hosts https://localhost:9200 \
+  --elasticsearch.ssl.certificateAuthorities=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/ca/ca.crt \
+  --elasticsearch.ssl.certificate=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/instance/instance.crt \
+  --elasticsearch.ssl.key=/home/elastic/elasticsearch-${EK_VERSION}/config/certs/instance/instance.key \
```

Comment:
I'd be surprised by this on why Kibana requires Elasticsearch's private key here (please ignore this comment, it's just my feeling :) )